From ab7dbc798dff2b49f5f291edeac1cded4e82d491 Mon Sep 17 00:00:00 2001
From: Rashed Abdel-Tawab <rashed@linux.com>
Date: Sat, 1 Oct 2016 20:33:00 -0400
Subject: [PATCH 15/39] extract_utils: Add ability to set custom vendor.mk name

In some circumstances we might want to set a custom vendor.mk
filename. If not set, then just default to the current device name

Change-Id: Icb59ebee67ce17b41ac613020685773efb01103d
---
 build/tools/extract_utils.sh | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/build/tools/extract_utils.sh b/build/tools/extract_utils.sh
index a4197a53..771e6bfb 100644
--- a/build/tools/extract_utils.sh
+++ b/build/tools/extract_utils.sh
@@ -48,6 +48,7 @@ trap cleanup EXIT INT TERM ERR
 # $3: CM root directory
 # $4: is common device - optional, default to false
 # $5: cleanup - optional, default to true
+# $6: custom vendor makefile name - optional, default to false
 #
 # Must be called before any other functions can be used. This
 # sets up the internal state for a new vendor configuration.
@@ -76,7 +77,12 @@ function setup_vendor() {
         mkdir -p "$CM_ROOT/$OUTDIR"
     fi
 
-    export PRODUCTMK="$CM_ROOT"/"$OUTDIR"/"$DEVICE"-vendor.mk
+    VNDNAME="$6"
+    if [ -z "$VNDNAME" ]; then
+        VNDNAME="$DEVICE"
+    fi
+
+    export PRODUCTMK="$CM_ROOT"/"$OUTDIR"/"$VNDNAME"-vendor.mk
     export ANDROIDMK="$CM_ROOT"/"$OUTDIR"/Android.mk
     export BOARDMK="$CM_ROOT"/"$OUTDIR"/BoardConfigVendor.mk
 
-- 
2.15.0

